examples/shader/compute_shader_game_of_life.rs

```diff
@@ -94,29 +121,77 @@ impl Plugin for GameOfLifeComputePlugin {
     }
 }
 
-#[derive(Resource, Clone, Deref, ExtractResource, AsBindGroup)]
-struct GameOfLifeImage {
-    #[storage_texture(0, image_format = R32Float, access = ReadWrite)]
-    texture: Handle<Image>,
+#[derive(Resource, Clone, ExtractResource)]
+struct GameOfLifeImages {
+    texture_a: Handle<Image>,
+    texture_b: Handle<Image>,
+}
+
+// Manual implementation of AsBindGroup instead of the derive
+// The two textures have the same role and will be swapped in the bind group
+// In the layout, one is `ReadOnly`, the other is `WriteOnly`
+impl AsBindGroup for GameOfLifeImages {
+    type Data = ();
+
+    fn label() -> Option<&'static str> {
+        Some("GameOfLifeImages")
+    }
+
+    fn unprepared_bind_group(
+        &self,
+        _layout: &BindGroupLayout,
+        _render_device: &RenderDevice,
+        _images: &RenderAssets<Image>,
+        _fallback_image: &bevy::render::texture::FallbackImage,
+    ) -> Result<UnpreparedBindGroup<Self::Data>, AsBindGroupError> {
+        Ok(UnpreparedBindGroup {
+            bindings: vec![],
+            data: (),
+        })
+    }
+
+    fn bind_group_layout_entries(_render_device: &RenderDevice) -> Vec<BindGroupLayoutEntry>
+    where
+        Self: Sized,
+    {
+        vec![
+            texture_storage_2d(TextureFormat::R32Float, StorageTextureAccess::ReadOnly)
+                .build(0, ShaderStages::COMPUTE),
+            texture_storage_2d(TextureFormat::R32Float, StorageTextureAccess::WriteOnly)
+                .build(1, ShaderStages::COMPUTE),
+        ]
+    }
 }
 
 #[derive(Resource)]
-struct GameOfLifeImageBindGroup(BindGroup);
+struct GameOfLifeImageBindGroups([BindGroup; 2]);
 
 fn prepare_bind_group(
     mut commands: Commands,
     pipeline: Res<GameOfLifePipeline>,
     gpu_images: Res<RenderAssets<Image>>,
-    game_of_life_image: Res<GameOfLifeImage>,
+    game_of_life_images: Res<GameOfLifeImages>,
     render_device: Res<RenderDevice>,
 ) {
-    let view = gpu_images.get(&game_of_life_image.texture).unwrap();
-    let bind_group = render_device.create_bind_group(
+    let view_a = gpu_images.get(&game_of_life_images.texture_a).unwrap();
+    let view_b = gpu_images.get(&game_of_life_images.texture_b).unwrap();
+    let bind_group_0 = render_device.create_bind_group(
+        None,
+        &pipeline.texture_bind_group_layout,
+        &BindGroupEntries::sequential((
+            BindingResource::TextureView(&view_a.texture_view),
+            BindingResource::TextureView(&view_b.texture_view),
+        )),
+    );
```

Comment:
```suggestion
    let bind_group_0 = render_device.create_bind_group(
        None,
        &pipeline.texture_bind_group_layout,
        &BindGroupEntries::sequential((&view_a.texture_view, &view_b.texture_view)),
    );
```

You don't need to wrap it in a `BindingResource::TextureView`